internal/query/project_metadata.go

```diff
@@ -0,0 +1,241 @@
+package query
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"time"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/zitadel/logging"
+
+	"github.com/zitadel/zitadel/internal/api/authz"
+	"github.com/zitadel/zitadel/internal/eventstore/handler/v2"
+	"github.com/zitadel/zitadel/internal/query/projection"
+	"github.com/zitadel/zitadel/internal/telemetry/tracing"
+	"github.com/zitadel/zitadel/internal/zerrors"
+)
+
+type ProjectMetadataList struct {
+	SearchResponse
+	Metadata []*ProjectMetadata
+}
+
+type ProjectMetadata struct {
+	CreationDate  time.Time
+	ChangeDate    time.Time
+	ProjectID     string
+	ResourceOwner string
+	Sequence      uint64
+	Key           string
+	Value         []byte
+}
+
+type ProjectMetadataSearchQueries struct {
+	SearchRequest
+	Queries []SearchQuery
+}
+
+var (
+	projectMetadataTable = table{
+		name:          projection.ProjectMetadataProjectionTable,
+		instanceIDCol: projection.ProjectMetadataColumnInstanceID,
+	}
+	ProjectMetadataProjectIDCol = Column{
+		name:  projection.ProjectMetadataColumnProjectID,
+		table: projectMetadataTable,
+	}
+	ProjectMetadataCreationDateCol = Column{
+		name:  projection.ProjectMetadataColumnCreationDate,
+		table: projectMetadataTable,
+	}
+	ProjectMetadataChangeDateCol = Column{
+		name:  projection.ProjectMetadataColumnChangeDate,
+		table: projectMetadataTable,
+	}
+	ProjectMetadataResourceOwnerCol = Column{
+		name:  projection.ProjectMetadataColumnResourceOwner,
+		table: projectMetadataTable,
+	}
+	ProjectMetadataInstanceIDCol = Column{
+		name:  projection.ProjectMetadataColumnInstanceID,
+		table: projectMetadataTable,
+	}
+	ProjectMetadataSequenceCol = Column{
+		name:  projection.ProjectMetadataColumnSequence,
+		table: projectMetadataTable,
+	}
+	ProjectMetadataKeyCol = Column{
+		name:  projection.ProjectMetadataColumnKey,
+		table: projectMetadataTable,
+	}
+	ProjectMetadataValueCol = Column{
+		name:  projection.ProjectMetadataColumnValue,
+		table: projectMetadataTable,
+	}
+)
+
+func (q *Queries) GetProjectMetadataByKey(ctx context.Context, shouldTriggerBulk bool, projectID string, key string, withOwnerRemoved bool, queries ...SearchQuery) (metadata *ProjectMetadata, err error) {
```

Comment:
You can remove `withOwnerRemoved` as it is not used.

Reply:
Done